### PR TITLE
Processing - Output tables with no geometry througt OutputVector

### PR DESCRIPTION
**Pull request by arnaud-morvan**
Here I add possibility to create .dbf, .csv, .ods and .xlsx files, memory layers or spatialite/postgis tables without geometry in OutputVector class.

This is very helpful for `Refactor field` algorithm, and possibly for all attributes algorithms in Vector table tools category, as we can process all tables, even if they do not have geometry.

These features were funded by: MEDDE (French Ministry of Sustainable Development)

### CODE REVIEW

**Comment by nyalldawson** [comment]
Why do the memory, PostGIS and spatialite outputs not support geometryless tables? The providers themselves have support for this.

**Comment by arnaud-morvan** [comment]
Thanks for review.
I was facing errors with those providers, now this is resolved, by putting geometry column name to ``None`` in datasource uri.

File: python/plugins/processing/core/outputs.py [hunk 1]
```diff
@@ -250,15 +250,34 @@ class OutputVector(Output):
...
+        Output.__init__(self, name, description, hidden)
+        self.base_input = base_input
+        self.base_layer = None
+
+    def hasGeometry(self):
+        if self.base_layer is None:
+            return True
+        return dataobjects.canUseVectorLayer(self.base_layer, [-1])
+
+    def getSupportedExtensions(self):
```
Comment (volaya) [inline python/plugins/processing/core/outputs.py]: This method name might be confusing, considering that the extensions it returns are only for vector layers and tables, but not for rasters...

Comment (arnaud-morvan) [inline python/plugins/processing/core/outputs.py]: Note that the class name is OutputVector

Comment (alexbruy) [inline python/plugins/processing/core/outputs.py]: I'm with Victor here, better to be consistent with other classes/packages. We have `getSupportedOutputRasterLayerExtensions`, `getSupportedOutputVectorLayerExtensions` and `getSupportedOutputTableExtensions` in other places

Comment (arnaud-morvan) [inline python/plugins/processing/core/outputs.py]: I've changed the method name, but, for me, this is do not really make sense with the polymorphism concept.

**Comment by arnaud-morvan** [comment]
@volaya @alexbruy 
I would like it to be merge in master before freeze on next friday.
This bring good feature to process tables without geometry.
Note that I've already tested it with modeler, command line and bacth dialog.

**Pull request merged**